src/OpenSSL/SSL.py

```diff
@@ -161,6 +164,53 @@ class SysCallError(Error):
     pass
 
 
+_CFFI_VERSION_RE = re.compile(r"^(\d+)\.(\d+).*$")
+
+
+def _make_buffer_checker(cffi_version_string):
+    """
+    Returns a callable that converts a sendable buffer object into something
+    that can be sent by OpenSSL using CFFI. Has different supported types
+    depending on the CFFI version.
+
+    :param cffi_version_string: The version of CFFI in use, from
+        ffi.__version__.
+    """
+    cffi_version = tuple(
+        int(component) for component in
+        _CFFI_VERSION_RE.match(cffi_version_string).group(1, 2)
```

Comment:
Why not just `from packaging.version import parse`? Having pyOpenSSL depend on packaging (which is a dependency of both setuptools and cryptography) doesn't seem unreasonable.